app/views/pets/index.html.erb

```diff
@@ -24,6 +24,16 @@
       <p>Sex: <%= pet.sex %><p>
       <p>Shelter: <a class="shelter-link hover" href="/shelters/<%= pet.shelter_id %>"><%= pet.shelter.name %></a></p>
 
+      <% if @favorite.contents[pet.id.to_s].nil? || @favorite.contents[pet.id.to_s] == 0 %>
+      <section id="fav-<%= pet.id %>" >
+          <%= button_to "Add #{pet.name} to favorites", "/favorites/#{pet.id}?page=index", method: :patch, class: "grey hover" %>
+      </section><br>
```

Comment:
more query params (?page=index) in the path. they don't effect the routes, but they get passed into `params` so we get a bit more info into the controller